install.sh

```diff
@@ -25,19 +25,30 @@ yarn_get_tarball() {
   else
     url=https://yarnpkg.com/latest.tar.gz
   fi
-  # Get both the tarball and its GPG signature
+
+  download_to_tmp() {
+    if curl --fail -L -o "$1" "$2"; then
+      printf "$green> Downloaded $2.$reset\n"
+    else
+      printf "$red> Failed to download $2.$reset\n"
+      exit 1;
+    fi
+  }
+
   tarball_tmp=`mktemp -t yarn.tar.gz.XXXXXXXXXX`
-  if curl --fail -L -o "$tarball_tmp#1" "$url{,.asc}"; then
```

Comment:
Yeah I tried it that was first of all, but the problem is that the second file doesn't get saved to the temp file. See;
```
/ # curl --fail -L -o "$tarball_tmp#1" "$url" "$url.asc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    92  100    92    0     0     30      0  0:00:03  0:00:02  0:00:01   190
100   595    0   595    0     0     98      0 --:--:--  0:00:06 --:--:--  1126
100 3537k  100 3537k    0     0   285k      0  0:00:12  0:00:12 --:--:--  915k
-----BEGIN PGP SIGNATURE-----

iQIzBAABCAAdFiEEagEMUWYAZZmqF/CBRsITDf0kl/UFAlhVL+QACgkQRsITDf0k
l/V1WA/+LkAyUUaW3hYLfrznPtqNOcnQGcgdYBqPPgNkZ5teHk9/Xjn1etpoA+Za
1aMETfPLkVxm9wlMrRERtQvWNmLm3cnxCrrPAAvoGQMw9nfNqkbDUWWVeYZwb3R8
C9Hcl/CSvl0rfl1rcBL6YCSkdguo92L1ENjXFLTaNx/0sXhNcvSeRV9PaP1xlJde
Vp+/SAs/Wh60H2oqGRyXN/8e7qhbCpuxsVXrUHfzcZJ4hNfDhnZ3/YRSGn29MTZa
lpDU53zpvwgQT8NpcMLdgafCuNnocTrcMdtOaoxvfTs9cROVu3e5lQh1u9ALFLu4
VJRO8rVVoyTzh7NyFOJT1NxLgFMji/5yUemm92dAEN1fIzG6ZGrEk8Q4H7o0XsoR
CkOc275X5d5JWmmBn0Nitk5FX9RLzYVCTGkKjg5cHZeF1wu0ouOwQ1kYhtBPzB3s
LXGbH9ZMRV6UgZZHo9rIz0V7OyQv3tzQJPMUyWNXnysN0LYDY6+wXv3wI22f0yr9
YHrI0r4oEjrpQzake54okQl4hon+sdQ1iUd/jkN52mz3CFoqWwlqq02iHt19WGs0
yFRmfI6glOYew9DWZOADejITvOgDRlSTElju3nSvLkkzs9Hd3AX43aVodJonPfBT
YE8FvWUPYwUSirz/ABLTDcVAz1G4wXzWVE4M9E7DxVmM+JJTJIc=
=5/GI
-----END PGP SIGNATURE-----/
```

I decided on having it split as a function for readability would make more sense?